lessons/flow.md

```diff
@@ -112,21 +149,207 @@ No errors!
 
 Congratulations, you are ready to develop using Flow 🚀.
 
-### Prepare your files for Flow 
+### 2.3 Prepare your files for Flow 
 
 In order to tell the compiler which file it should check, you need to mark it as a Flow file by placing the following line at the top of the file:
 
 ```js
 // @flow
 ```
 
-Then the compiler will analise all these files at compile time to ensure consistency across the type definitions. 
+Then the compiler will analyse all these files at compile time to ensure consistency across the type definitions. 
 
 ## 3. Writing Flow code
+
+### 3.1 [Primitive types](https://flow.org/en/docs/types/primitives/)
+
+#### 3.1.1 Literal vs. wrapper types
+
+In Flow there are types for the literal values that are lowercase, and types for the wrapper objects which are 
+capitalized. Therefore, the type `Number` is not the same as the type `number`. The full list of primitive types is:
+
+* [`Booleans`](https://flow.org/en/docs/types/primitives/#toc-booleans)
+* [`Number`](https://flow.org/en/docs/types/primitives/#toc-numbers)
+* [`String`](https://flow.org/en/docs/types/primitives/#toc-strings)
+
+#### 3.1.2 [`null` vs. `void`](https://flow.org/en/docs/types/primitives/#toc-null-and-void)
+
+The type of `null` is `null` but the type of `undefined` is `void`. Therefore, a function which input is a number and 
+returns `undefined`, its type will be `(number) => void`.
+
+#### 3.1.3 [Maybe types](https://flow.org/en/docs/types/primitives/#toc-maybe-types)
+
+In order to define a maybe type, you need put a `?` before the type. In this way, the type defined, `void` and `null` 
+will be accepted.
+
+For example: 
+```flow js
+function foo (x: ?number) {
+  //....
+}
+
+```
+It will accept any number, `null` or `undefined` as its input. 
+
+#### 3.1.4 Optional [object parameters](https://flow.org/en/docs/types/primitives/#toc-optional-object-properties) and [function parameters](https://flow.org/en/docs/types/primitives/#toc-optional-function-parameters)
+
+It is possible to declare a object property of a function argument by placing the `?` right after the variable name.
+
+```flow js
+function foo(value?: string) {
```

Comment:
I've never seen this usage of optional object properties as being used at the top level of a parameter list (not that it doesn't perhaps happen)

As per the docs I think it's more often used for properties in object parameters rather than parameters themselves:
```js
function acceptsObject(value: { foo?: string }) {
  //...
}
```